YAML-конфиг

Oreolek opened the pull request:
Рефактор на общий конфиг
Много исправлений

Oreolek committed:
прояснил установку

Oreolek committed:
Merge branch 'master' of https://github.com/Himura2la/Cosplay2-Downloader

Oreolek committed:
Merge branch 'cr17' of https://github.com/Himura2la/Cosplay2-Downloader

Oreolek committed:
Merge branch 'cr17' of https://github.com/Himura2la/Cosplay2-Downloader

Oreolek committed:
Merge branch 'cr17' of https://github.com/Himura2la/Cosplay2-Downloader

Oreolek committed:
Merge branch 'master' of https://github.com/Himura2la/Cosplay2-Downloader

Oreolek committed:
Удалил BAT-ники

Oreolek committed:
Конфиг на YAML

Oreolek committed:
Добавил EditorConfig

Oreolek committed:
Пароль тоже в конфиге

Oreolek committed:
checker YAML config

Oreolek committed:
Убрал compare_folders (просто используйте diff)

Oreolek committed:
Удалил sql_query (просто откройте sqlite)

Oreolek committed:
Удалил regex_renamer (просто установите rename)

Oreolek committed:
get_data + YAML

Oreolek committed:
get_files + YAML

Oreolek committed:
Пароль в конфиге для get_data

Oreolek committed:
Получить пароль с консоли, если его нет в конфиге

Oreolek committed:
Необязательные скрипты в папке etc/

Oreolek committed:
описание для папки etc

Oreolek committed:
больше описаний

Oreolek committed:
mktex + YAML

Oreolek committed:
Варианты скачивателей файлов

Himura2la committed:
Меньше файлов больше описания

Himura2la committed:
Merge branch 'master' into Oreolek-master

Himura2la commented:
@Oreolek, осторожно, я запушил, советую взяться ))

Himura2la commented on .editorconfig:
```diff
@@ -0,0 +1,11 @@
+
+# Unix-style newlines with a newline ending every file
+[*]
+end_of_line = lf
```
А у тех кто на винде стоит автоконверт на `crlf` в гите, и для них это не справедливо. 
А еще, кодировку тут тоже неплохо было бы настроить

Oreolek replied:
Можно, см. http://editorconfig.org/
Я не в курсе как интерпретатор Python относится к `crlf`

Himura2la replied:
на винде это стандарт. А на никсы `crlf` не пройдет. Я к тому, что может не настраивать line endings чтобы не сбивать редакторы на винде, которые считают этот файл истиной в последней инстанции

Добавь тогда

```
charset = utf-8
```

если будешь редактировать, чтобы не плодить мёржей

Himura2la committed:
вернул благодарности и всё такое

Oreolek committed:
Merge branch 'master' of https://github.com/Himura2la/Cosplay2-Downloader

Oreolek committed:
Merge branch 'master' of github.com:Oreolek/Cosplay2-Downloader

Himura2la commented:
@Oreolek, включи плиз спелчекер, чтобы он подчеркивал тебе идентификаторы из слитных слов и бесил ))
Это действительно сложно читать. 
![image](https://user-images.githubusercontent.com/7802808/40226726-2847d450-5a95-11e8-9bf1-3ff7803309aa.png)

А еще, постарайся плиз как можно меньше использовать `+` для конкатенации строк, форматированные строки лучше. 

Вот например тут

https://github.com/Oreolek/Cosplay2-Downloader/blob/5e808335699d6d057ccc753b3d89c21e788f7369/get_files.py#L192

лучше так

```
non_scene = ','.join(['"%s"' % s for s in config['non_scene']])
```

Я могу и сам это всё поправить когда закончишь с изменениями

Oreolek committed:
Editorconfig - кодировка и без указания CR или LF

Oreolek committed:
query select

Oreolek committed:
Мелкие правки mktex.py

Oreolek committed:
Не допускать двойных пробелов в названиях файлов

На двойных пробелах ломается TeX.

Oreolek committed:
abort on malformed SQL

Oreolek committed:
Выковыривание номеров

- Искать номер по регулярке
- Брать номер заявки вместо номера голосования (если NULL)
- Пропускать пустые папки

Oreolek committed:
config

Oreolek committed:
Опция генерировать пути сразу для PDF

Himura2la commented:
@Oreolek , я все читаю, если что ))
Приму после вашего феста, чтобы дальше уже все свои правки вносить независимот от твоей репы )

Oreolek committed:
Обрабатывать все фото, а не первое с заявки

Oreolek committed:
отменить PDF по умолчанию

Oreolek committed:
explicit if

Oreolek committed:
Не учитывать ориентацию квадратных фото

Это две разных переменных

Oreolek committed:
Оптимизация под новый print.tex

Oreolek committed:
Пропускать референсы персонажей при генерации

Oreolek committed:
Использовать сокращения категорий из базы

Oreolek committed:
Обновлённый mktex для проставленных номеров заявок

Oreolek committed:
Правильное заполнение TeX

Oreolek committed:
Хранить URL феста тоже в конфиге

Oreolek committed:
Обработка особых символов

Oreolek committed:
Замена «шапочек»

Oreolek committed:
renewcommand вместо newcommand

Himura2la commented:
надеюсь, я в этом смогу разобраться теперь ))
